### PR TITLE
Change the doc to the new format

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

File: doc/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+To be able to differentiate which pod responds to our request we add a `podName` variable to the response object. `Application.java` and `CommandController.java` also reads the `podName` as follows:
+
+[source,java]
+----
+    @Value("#{environment.MY_POD_NAME}")
+    private String podName;
+----
+
+We also add the `kubernetes.yaml` file will create two Micronaut applications and a service to connect them to outer world.
```
Comment (mtyazici) [inline doc/modules/ROOT/pages/index.adoc:85]:
```suggestion
We also add the `kubernetes.yaml` file which will create two Micronaut applications and a service to connect them to outer world.
```

File: doc/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+[source,yaml]
+----
+containers:
+  name: hazelcast-micronaut-container
+    image: springboot-tomcat-session-replication-on-kubernetes:0.1.0
+    imagePullPolicy: IfNotPresent
+    ports:
+----
+
+[NOTE]
```
Comment (mtyazici) [inline doc/modules/ROOT/pages/index.adoc:109]: Here I don't know how to make it pretty. It would be good if we could split the note to two parts.

File: doc/antora.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+name: springboot-tomcat-session-replication-on-kubernetes    # a brief module name. This will be included in the guide url.
+title: Guide Title  # a brief module title. This will not show up on the website.
```
Comment (mtyazici) [inline doc/antora.yml:2]:
```suggestion
title: Spring Boot Tomcat Session Replication on Kubernetes using Hazelcast
```

File: doc/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
+
+:github-address: https://github.com/hazelcast-guides/springboot-tomcat-session-replication-on-kubernetes
+:templates-url: https://raw.githubusercontent.com/hazelcast-guides/adoc-templates/master
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:3]: Already have `:templates-url:` below 
```suggestion
```

File: doc/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+
+To be able to differentiate which pod responds to our request we add a `podName` variable to the response object. `Application.java` and `CommandController.java` also reads the `podName` as follows:
+[source,java]
+----
+    @Value("#{environment.MY_POD_NAME}")
+    private String podName;
+----
+
+We also add the `kubernetes.yaml` file which will create two Micronaut applications and a service to connect them to outer world.
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:70]: Micronaut?

File: doc/modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+== Create a Docker Image
+
+In order to create the Docker image of the application, we will use the https://github.com/GoogleContainerTools/jib[Jib] tool, and its related Maven plugin. It allows to build containers from Java applications without a Docker file or even changing the pom.xml file. To build the image, run the command below:
+[source,bash]
+----
+$ mvn clean compile com.google.cloud.tools:jib-maven-plugin:1.8.0:dockerBuild
+----
+
+This command will create a docker image named `springboot-tomcat-session-replication-on-kubernetes:0.1.0`. If you want to change the name to push it to some registry, you can run the following instead:
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:80]: Sorry, I'm a little bit obsessed with this capital issue.
```suggestion
This command will create a Docker image named `springboot-tomcat-session-replication-on-kubernetes:0.1.0`. If you want to change the name to push it to some registry, you can run the following instead:
```

File: doc/modules/ROOT/pages/index.adoc (hunk 2 above)
Comment (enozcan) [reply doc/modules/ROOT/pages/index.adoc:96]: Just include another NOTE section. e.g.
```
[NOTE]
====
note1
====

[NOTE]
====
note2
====
```

File: doc/modules/ROOT/pages/index.adoc [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+$ docker login
+$ docker push YOUR-NAME/springboot-tomcat-session-replication-on-kubernetes:0.1.0
+----
+====
+
+== Configure RBAC
+
+https://github.com/hazelcast/hazelcast-kubernetes[Hazelcast Kubernetes discovery plugin] makes calls to Kubernetes API to provide automatic member discovery. Therefore, it needs to have specific ClusterRole rules granted. You can apply the minimal RBAC configuration (for the `default` service account in the `default` namespace) with the following command.
+
+By this time make sure you have your kubernetes environment running.
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:120]:
```suggestion
By this time make sure you have your Kubernetes environment running.
```

File: doc/modules/ROOT/pages/index.adoc [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+----
+
+[NOTE]  
+====
+- If you use service account other than `default` or namespace other than `default`, you need to modify `rbac.yaml` file.
+- If your Kubernetes cluster does not use RBAC, you can skip the "Configure RBAC" step
+====
+
+
+== Deploy Micronaut Application to Kubernetes
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:133]: Left over content from Micronaut

File: doc/modules/ROOT/pages/index.adoc [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+$ kubectl get pods
+----
+
+== Testing the Application
+
+Launch a curl container inside kubernetes cluster. This will create an interactable shell.
+
+[source,bash]
+----
+$ kubectl run curl --rm --image=radial/busyboxplus:curl -i --tty
```
Comment (enozcan) [inline doc/modules/ROOT/pages/index.adoc:152]: I wonder why did you change the testing logic and a few others from the original README.md content.

File: doc/modules/ROOT/pages/index.adoc (hunk 5 above)
Comment (mtyazici) [reply doc/modules/ROOT/pages/index.adoc:70]:
```suggestion
We also add the `kubernetes.yaml` file which will create two Spring Boot applications and a service to connect them to outer world.
```

File: doc/modules/ROOT/pages/index.adoc [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+After this command you will be able to interact with Minikube’s Docker daemon, so the docker image built by `mvn clean compile..` will be accessible by Minikube's Docker daemon. Remember this is only valid for the shell that ran the commmand.
+
+If you are not using `Minikube` and you ran the  `mvn clean compile ...` command changing the image name with your Docker Hub ID. You can push the Docker image to your registry in Docker Hub with the following commands:
+
+[source,bash]
+----
+$ docker login
+$ docker push YOUR-NAME/springboot-tomcat-session-replication-on-kubernetes:0.1.0
+----
+====
```
Comment (mtyazici) [inline doc/modules/ROOT/pages/index.adoc:114]:
```suggestion
If you are using `Minikube` start the Minikube and run the following command before the `mvn clean compile ...` command. 
[source,bash]
----
$ eval $(minikube docker-env)
----
After this command you will be able to interact with Minikube’s Docker daemon, so the docker image built by `mvn clean compile..` will be accessible by Minikube's Docker daemon. Remember this is only valid for the shell that ran the commmand.
====

[NOTE]
====
If you are not using `Minikube` and you ran the  `mvn clean compile ...` command changing the image name with your Docker Hub ID. You can push the Docker image to your registry in Docker Hub with the following commands:

[source,bash]
----
$ docker login
$ docker push YOUR-NAME/springboot-tomcat-session-replication-on-kubernetes:0.1.0
----
====
```

File: doc/modules/ROOT/pages/index.adoc (hunk 8 above)
Comment (mtyazici) [reply doc/modules/ROOT/pages/index.adoc:133]:
```suggestion
== Deploy The Application to Kubernetes
```

File: doc/modules/ROOT/pages/index.adoc (hunk 9 above)
Comment (mtyazici) [reply doc/modules/ROOT/pages/index.adoc:154]: In other guides I used this testing logic, so it is easier to follow each guide if we use the same one. I don't understand what  you mean by "a few others".

Comment (enozcan) [reply doc/modules/ROOT/pages/index.adoc:166]: I tried this and actually it looks better than the original one - no need for cluster ip at all. What I meant by others were the following curl commands. However, this is OK now.

**Comment by enozcan** [comment]
Thanks @mtyazici! You're doing a valuable job for the guides website. I have an advice for your incoming works:
I see many leftovers from other guide contents when you create a new one (i.e. Micronaut keyword in Spring guide). If you think a particular section is common among more than one guide, then consider creating a template for this section and include in https://github.com/hazelcast-guides/adoc-templates with proper guide dependent variables (e.g. framework, request url, request port, etc.). This will make creating new guides faster.

**Comment by mtyazici** [comment]
Yeah I will try using templates for the next guides. At the time I was working with multiple guides, so sometimes I got confused which one I was doing. Next ones should be cleaner overall.
